modules/xml/src/smithy4s/xml/XmlDocument.scala

```diff
@@ -76,6 +82,14 @@ object XmlDocument {
     }
   }
 
+  private def getRootName[A](schema: Schema[A]): XmlQName = {
+    schema.hints
+      .get(XmlName)
+      .map(_.value)
+      .map(XmlQName.parse)
```

Comment:
totally nitpicky suggestion: extract these to a val so it's clearer that `getOrElse` is a fallback for the whole thing